File: backend/utils/severity.py
"""
utils/severity.py — Convert numeric severity (1-10) to labels and colours.
"""


SEVERITY_MAP = {
    (1, 2): ("Info",     "white",  "#FFFFFF"),
    (3, 4): ("Minor",    "blue",   "#378ADD"),
    (5, 6): ("Moderate", "yellow", "#DDB929"),
    (7, 8): ("Serious",  "orange", "#EF9F27"),
    (9, 10): ("Critical","red",    "#E24B4A"),
}


def severity_label(score: int) -> str:
    for (lo, hi), (label, *_) in SEVERITY_MAP.items():
        if lo <= score <= hi:
            return label
    return "Unknown"


def severity_colour(score: int) -> str:
    for (lo, hi), (_, _, colour) in SEVERITY_MAP.items():
        if lo <= score <= hi:
            return colour
    return "#888780"


def severity_risk_level(score: int) -> str:
    return severity_label(score)


def should_notify_authorities(score: int) -> bool:
    """Return True when score reaches 'Serious' or above."""
    return score >= 7


def crowd_threshold_for_severity_raise(current_severity: int) -> int:
    """
    How many crowd confirmations are needed to raise severity by 1.
    Higher current severity requires more confirmations.
    """
    if current_severity <= 4:
        return 2
    if current_severity <= 7:
        return 3
    return 5

File: frontend/src/utils/severity.js
/**
 * utils/severity.js
 * ─────────────────
 * Mirror of backend utils/severity.py for client-side use.
 */

export const SEVERITY_LEVELS = [
  { range: [9, 10], label: 'Critical', color: '#E24B4A', bg: 'bg-red-900/50',    text: 'text-red-400'    },
  { range: [7, 8],  label: 'Serious',  color: '#EF9F27', bg: 'bg-orange-900/40', text: 'text-orange-400' },
  { range: [5, 6],  label: 'Moderate', color: '#DDB929', bg: 'bg-yellow-900/40', text: 'text-yellow-400' },
  { range: [3, 4],  label: 'Minor',    color: '#378ADD', bg: 'bg-blue-900/40',   text: 'text-blue-400'   },
  { range: [1, 2],  label: 'Info',     color: '#888780', bg: 'bg-gray-800',      text: 'text-gray-400'   },
]

export function getSeverityLevel(score) {
  return SEVERITY_LEVELS.find(l => score >= l.range[0] && score <= l.range[1]) ?? SEVERITY_LEVELS[4]
}

export function severityLabel(score)  { return getSeverityLevel(score).label }
export function severityColor(score)  { return getSeverityLevel(score).color }

export function incidentEmoji(type) {
  const map = {
    'Fire':              '🔴',
    'Flood':             '🔵',
    'Medical':           '🟢',
    'Medical emergency': '🟢',
    'Accident':          '🟠',
    'Security':          '🟡',
    'Building collapse': '🔴',
    'Power outage':      '⚪',
  }
  return map[type] ?? '⚫'
}
